apache/flink
# [FLINK-3587] Bump Calcite version to 1.7.0

- [X] General
  - The pull request references the related JIRA issue
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message

  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [X] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

This pull request updates the Apache Calcite dependency to 1.7. With Calcite 1.7 a few method APIs and the some behavior changed.

- Fix changed API for optimizer cost and cardinality estimates.
- Add DataSetValues and DataStreamValues due to changed Calcite RelNode generation.
- Pass TableEnvironment instead of TableConfig for DataSet and DataStream translation.
- Add methods to create new DataSources to BatchTableEnvironment and StreamTableEnvironment.
- Remove copied Calcite rule that got fixed.
